src/http.rs

```diff
@@ -117,30 +122,51 @@ impl HttpClient {
             .post(&url)
             .headers(self.headers.clone())
             .body(s)
-            .send()?
-            .text()?;
-
-        let edn_resp = Edn::from_str(&resp.replace("#inst", ""));
-        edn_resp.or(Ok(edn!({:status ":internal-server-error", :code 500})))
+            .send()?;
+
+        if resp.status().as_u16() < 300 {
+            let resp_body = resp.text()?;
+            let edn_resp = Edn::from_str(&resp_body.replace("#inst", ""));
+            edn_resp.or(Err(CruxError::BadRequestError(format!(
+                "entity-timed responded with {} for id \"{}\" ",
+                500, crux_id
+            ))))
+        } else {
+            Err(CruxError::BadRequestError(format!(
+                "entity-timed responded with {} for id \"{}\" ",
+                resp.status().as_u16(),
+                crux_id
+            )))
+        }
     }
 
     /// Function `entity_tx` requests endpoint `/entity-tx` via `POST` which retrieves the docs and tx infos
     /// for the last document for that ID saved in CruxDB.
     pub fn entity_tx(&self, id: CruxId) -> Result<EntityTxResponse, CruxError> {
+        let crux_id = edn_rs::to_string(id);
+
         let mut s = String::new();
         s.push_str("{:eid ");
-        s.push_str(&edn_rs::to_string(id));
+        s.push_str(&crux_id);
         s.push_str("}");
 
         let resp = self
             .client
             .post(&format!("{}/entity-tx", self.uri))
             .headers(self.headers.clone())
             .body(s)
-            .send()?
-            .text()?;
-
-        EntityTxResponse::from_str(&resp.replace("#inst", ""))
+            .send()?;
+
+        if resp.status().as_u16() < 300 {
```

Comment:
Shoudn't we create a function to aggregate this `if`/`else` logic? :thinking: 
It seems the same every time.

Reply:
There are 4 `if` that differ in 2 different responses, also it is not easy to deal with all the borrowing needed here. I prefer to leave the ifs, but you are welcome to try to extract it kkkkk